include/deal.II/base/function.h

```diff
@@ -347,135 +347,158 @@ class Function : public FunctionTime<Number>,
 
-  virtual void vector_value (const Point<dim> &p,
-                             Vector<Number>   &return_value) const;
+    virtual Number value (const Point<dim>   &p,
+                          const unsigned int  component = 0) const;
 
-  virtual void value_list (const std::vector<Point<dim> > &points,
-                           std::vector<Number>            &values,
-                           const unsigned int              component = 0) const;
+    virtual void vector_value (const Point<dim> &p,
+                               Vector<Number>   &return_value) const;
 
-  virtual void vector_value_list (const std::vector<Point<dim> > &points,
-                                  std::vector<Vector<Number> >   &values) const;
+    virtual void value_list (const std::vector<Point<dim> > &points,
+                             std::vector<Number>            &values,
+                             const unsigned int              component = 0) const;
 
-  virtual Tensor<1,dim, Number> gradient (const Point<dim> &p,
-                                          const unsigned int component = 0) const;
+    virtual void vector_value_list (const std::vector<Point<dim> > &points,
+                                    std::vector<Vector<Number> >   &values) const;
 
-  virtual void vector_gradient (const Point<dim>            &p,
-                                std::vector<Tensor<1,dim, Number> > &gradients) const;
+    virtual Tensor<1,dim, Number> gradient (const Point<dim> &p,
+                                            const unsigned int component = 0) const;
 
-  virtual void gradient_list (const std::vector<Point<dim> > &points,
-                              std::vector<Tensor<1,dim, Number> >    &gradients,
-                              const unsigned int              component = 0) const;
+    virtual void vector_gradient (const Point<dim>            &p,
+                                  std::vector<Tensor<1,dim, Number> > &gradients) const;
 
-  virtual void vector_gradient_list (const std::vector<Point<dim> >            &points,
-                                     std::vector<std::vector<Tensor<1,dim, Number> > > &gradients) const;
-};
+    virtual void gradient_list (const std::vector<Point<dim> > &points,
+                                std::vector<Tensor<1,dim, Number> >    &gradients,
+                                const unsigned int              component = 0) const;
+
+    virtual void vector_gradient_list (const std::vector<Point<dim> >            &points,
+                                       std::vector<std::vector<Tensor<1,dim, Number> > > &gradients) const;
+  };
 
 
 
+  /**
+   * Provide a function which always returns the constant values handed to the
+   * constructor.
+   *
+   * Obviously, the derivatives of this function are zero, which is why we derive
+   * this class from <tt>ZeroFunction</tt>: we then only have to overload the
+   * value functions, not all the derivatives. In some way, it would be more
```

Comment:
here's the rationale for why we do the derivation. please remove this sentence as well if you change the inheritance structure.